lib/wda/iproxy.js

```diff
@@ -1,98 +1,33 @@
-import log from '../logger';
-import { killProcess } from './utils';
-import B from 'bluebird';
-import { logger } from 'appium-support';
-import { SubProcess } from 'teen_process';
-import { checkPortStatus } from 'portscanner';
-import { waitForCondition } from 'asyncbox';
-
-
-const IPROXY_STARTUP_TIMEOUT = 5000;
-
-const iproxyLog = logger.getLogger('iProxy');
+import { utilities } from 'appium-ios-device';
+import net from 'net';
 
 class iProxy {
-  constructor (udid, localport, deviceport, detached = true) {
-    this.expectIProxyErrors = true;
+  constructor (udid, localport, deviceport) {
     this.localport = parseInt(localport, 10);
     this.deviceport = parseInt(deviceport, 10);
     this.udid = udid;
-    this.iproxy = new SubProcess('iproxy', [localport, deviceport, udid], {
-      detached,
-      stdio: ['ignore', 'pipe', 'pipe'],
-    });
   }
 
-  async start () {
-    log.debug(`Starting iproxy to forward traffic from local port ${this.localport} ` +
-      `to device port ${this.deviceport} over USB for the device ${this.udid}`);
-    this.expectIProxyErrors = true;
-
-    return await new B((resolve, reject) => {
-      this.iproxy.on('exit', (code) => {
-        log.debug(`iproxy exited with code '${code}'`);
-        if (code) {
-          return reject(new Error(`iproxy exited with code '${code}'`));
-        }
-      });
-      this.iproxy.on('output', (stdout, stderr) => {
-        // do nothing if we expect errors
-        if (this.expectIProxyErrors) {
-          return;
-        }
-
-        let out = stdout || stderr;
-        for (let line of out.split('\n')) {
-          if (!line.length) {
-            continue;
-          }
-
-          if (line.includes('Resource temporarily unavailable')) {
-            // this generally happens when WDA does not respond,
-            // so print a more useful message
-            log.debug('Connection to WDA timed out');
-          } else {
-            iproxyLog.debug(line);
-          }
-        }
-      });
+  start () {
+    this.serverSocket = net.createServer(async (c) => {
+      try {
+        const socket = await utilities.connectPort(this.udid, this.deviceport);
+        socket.on('close', c.destroy);
+        c.on('close', socket.destroy);
+        c.pipe(socket);
+        socket.pipe(c);
+      } catch (e) {
+        c.destroy();
```

Comment:
shouldn't the exception be logged?

Reply:
Do you think it is that important? There are 2 possibilities: the device is missing or WDA is not up

Reply:
I saw issues where devices were disconnected unexpectedly and then it was not very clear why tests failed. It would be easier for us to debug such issues if we have a log

Reply:
Hmm. I need to handle the errors then. Need to think how to tackle that best

Reply:
This will now log nice errors